cmd/provenanced/cmd/genaccounts.go

```diff
@@ -506,6 +506,57 @@ enforced immediately.  An optional type flag can be provided or the default of C
 	return cmd
 }
 
+// AddGenesisCustomFloorPriceDenom returns add-genesis-msg-fee cobra command.
+func AddGenesisCustomFloorPriceDenom(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-genesis-custom-floor-denom [coin]",
+		Short: "Add a floor price denom and amount to genesis.json",
+		Long: `Add a floor price denom and amount to genesis.json. This will create a custom floor price denom and amount for calculating additional message costs.
+Currently, the denom and price defaults to 1905nhash
+		`,
+		Example: fmt.Sprintf(`$ %[1]s add-genesis-custom-floor-denom 0vspn`, version.AppName),
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			cdc := clientCtx.Codec
+			config := serverCtx.Config
+			config.SetRoot(clientCtx.HomeDir)
+			coin, err := sdk.ParseCoinNormalized(args[0])
+			if err != nil {
+				return fmt.Errorf("failed to parse coin: %w", err)
+			}
+			genFile := config.GenesisFile()
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+			msgFeesGenState := msgfeetypes.GetGenesisStateFromAppState(cdc, appState)
+
+			msgFeesGenState.Params.FloorGasPrice = coin
+
+			msgFeesGenStateBz, err := cdc.MarshalJSON(&msgFeesGenState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal msgfees genesis state: %w", err)
+			}
+
+			appState[msgfeetypes.ModuleName] = msgFeesGenStateBz
+
+			appStateJSON, err := json.Marshal(appState)
+			if err != nil {
+				return fmt.Errorf("failed to marshal application genesis state: %w", err)
+			}
+
+			genDoc.AppState = appStateJSON
+			return genutil.ExportGenesisFile(genDoc, genFile)
+		},
+	}
+	cmd.Flags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+	flags.AddQueryFlagsToCmd(cmd)
```

Comment:
interesting every other command on that file adds that ```// AddQueryFlagsToCmd adds common flags to a module query command.
func AddQueryFlagsToCmd(cmd *cobra.Command) {
	cmd.Flags().String(FlagNode, "tcp://localhost:26657", "<host>:<port> to Tendermint RPC interface for this chain")
	cmd.Flags().Int64(FlagHeight, 0, "Use a specific height to query state at (this can error if the node is pruning state)")
	cmd.Flags().StringP(tmcli.OutputFlag, "o", "text", "Output format (text|json)")

	// some base commands does not require chainID e.g `simd testnet` while subcommands do
	// hence the flag should not be required for those commands
	_ = cmd.MarkFlagRequired(FlagChainID)
}``` 🤔

Reply:
ok i think `AddTxFlagsToCmd` is more appropriate here.. tested with `make run`

Reply:
I had these confused with the `AddPaginationFlagsToCmd`. Still not sure if they're needed, but now, it at least makes more sense to me to have them.